### PR TITLE
Unique system ids for cached initdb

### PR DESCRIPTION
**Pull request by funbringer**
If enabled, generate unique system ids for nodes spawned from initdb cache.

### CODE REVIEW

File: testgres/utils.py [hunk 1]
```diff
@@ -68,6 +68,27 @@ def generate_app_name():
...
+    secs = int(date.timestamp())
+    usecs = date.microsecond
+
+    system_id = 0
+    system_id |= (secs << 32)
+    system_id |= (usecs << 12)
+    system_id |= (os.getpid() & 0xFFF)
+
+    # pack ULL in native byte order
+    return struct.pack('=Q', system_id)
```
Comment (arssher) [inline testgres/utils.py]: This will probably bail out with `struct.error: argument out of range` after 19 January 2038, but for now it is fine :)

**Comment by arssher** [comment]
I've modified shardman to use this stuff and it works great, thanks.

File: testgres/utils.py (hunk 1 above)
Comment (funbringer) [reply testgres/utils.py]: @arssher Why? We have 32 bits for seconds, and that's ~136 years.

Comment (arssher) [reply testgres/utils.py]: Ah, =Q is unsigned. Yes, then we have 68 years more.

**Pull request merged**